app/channels/application_cable/connection.rb

```diff
@@ -2,6 +2,9 @@ module ApplicationCable
   class Connection < ActionCable::Connection::Base
     identified_by :current_user
     def connect
+      if request.session[:auth_type] == 'remote'
```

Comment:
Very nice. In comparing this implementation with `MainController#login`, I note that the latter uses a `remote_auth?` helper, which checks another session key. I suspect it would be prudent to do the same here?

Reply:
Hi @david-yz-liu - thanks for the review! Sessions are not available in cable (hence the use of `request.session`), so we can't directly re-use `remote_auth?`. We do check the same session key here as in that method. Would you be alright with keeping it this way for simplicity, or is there another approach we should be thinking of?

Reply:
@pretendWhale oh sorry you're totally right, I missed that even though I was looking at exactly this line. All good :+1: